### PR TITLE
feat: dashboard UX improvements, help button, adoption badges, MFA guidance

### PR DESCRIPTION
**Pull request by CybotTM**
## Summary

Addresses user feedback from [NRNR-1532](https://jira.netresearch.de/browse/NRNR-1532):

- **Quick Start guide** on Dashboard for new installations (no passkeys yet) with step-by-step setup instructions and config status hints
- **Help button** in DocHeader (question-mark icon via ButtonBar API) so the Help tab is discoverable without the dropdown menu
- **MFA hint** on Dashboard suggesting admins can drop MFA requirement when passkeys are active
- **Adoption badges** (gamification): Getting started → Bronze → Silver → Gold → Platinum, always visible on adoption rate card
- **Enhanced Help page**: renamed MFA section to "Passkeys & MFA", added prominent "Are passkeys secure enough without MFA?" infobox
- **README**: Quick Start section, Passkeys & MFA guidance, TER docs link, rpId/rpName/origin in config table

### Cross-version compatibility

- Uses `InfoboxViewHelper::STATE_*` integer constants (works on v12/v13/v14) instead of `ContextualFeedbackSeverity` enum (v14 only as infobox state)
- Uses `enum_exists(IconSize::class)` runtime check with `'small'` string fallback for v12
- All UI elements follow TYPO3 backend conventions (`f:be.infobox`, `badge-*` classes, `card-container`, ButtonBar API)

## Test plan

- [ ] Dashboard shows "Getting started" card when no passkeys are registered
- [ ] Dashboard shows adoption badge (Getting started/Bronze/Silver/Gold/Platinum)
- [ ] Dashboard shows MFA hint when users have passkeys
- [ ] Help button (question-mark icon) visible in DocHeader on all views
- [ ] Help tab renders correctly with enhanced MFA section
- [ ] Verify on v12, v13, and v14 — no rendering errors
- [ ] PHPStan, unit tests, CGL pass

### CODE REVIEW

File: Classes/Controller/AdminModuleController.php [hunk 1]
```diff
@@ -77,13 +84,32 @@ public function dashboardAction(ServerRequestInterface $request): ResponseInterf
...
+        }
+
+        $adoptionPercentage = $stats->adoptionPercentage();
+
         $moduleTemplate->assignMultiple([
             'totalUsers' => $stats->totalUsers,
             'usersWithPasskeys' => $stats->usersWithPasskeys,
-            'adoptionPercentage' => $stats->adoptionPercentage(),
+            'adoptionPercentage' => $adoptionPercentage,
+            'adoptionBadge' => self::adoptionBadge($adoptionPercentage, $stats->totalUsers),
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Classes/Controller/AdminModuleController.php]: Fixed in 0a416bb. Changed to `$this->adoptionBadge()` (instance method).

File: Classes/Controller/AdminModuleController.php [hunk 2]
```diff
@@ -164,4 +191,43 @@ private function getEnforcementLevelOptions(): array
...
+            return ['label' => 'No users', 'class' => 'badge-secondary', 'icon' => 'actions-minus'];
+        }
+
+        return match (true) {
+            $percentage >= 100.0 => ['label' => 'Platinum', 'class' => 'badge-info', 'icon' => 'actions-bolt'],
+            $percentage >= 75.0 => ['label' => 'Gold', 'class' => 'badge-warning', 'icon' => 'actions-star'],
+            $percentage >= 50.0 => ['label' => 'Silver', 'class' => 'badge-secondary', 'icon' => 'actions-check'],
+            $percentage >= 25.0 => ['label' => 'Bronze', 'class' => 'badge-success', 'icon' => 'actions-arrow-up'],
+            default => ['label' => 'Getting started', 'class' => 'badge-danger', 'icon' => 'actions-rocket'],
+        };
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Classes/Controller/AdminModuleController.php]: Fixed in 0a416bb. Made non-static, using `$this->translate()` for all 6 badge labels. Added translation keys to `locallang.xlf`.

File: Classes/Controller/AdminModuleController.php [hunk 3]
```diff
@@ -77,13 +84,32 @@ public function dashboardAction(ServerRequestInterface $request): ResponseInterf
...
         }
 
+        $config = $this->configService->getConfiguration();
+        $hasEnforcement = false;
+        foreach ($groupData as $group) {
+            if ($group['enforcement'] !== EnforcementLevel::Off->value) {
+                $hasEnforcement = true;
+                break;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Classes/Controller/AdminModuleController.php]: Already fixed in 87ff488 (previous commit). Both `hasEnforcement` loop and `hasGroups` removed.

File: Resources/Private/Templates/AdminModule/Dashboard.html [hunk 4]
```diff
@@ -10,6 +10,51 @@ <h1><f:translate key="dashboard.title" extensionName="NrPasskeysBe" /></h1>
...
+                        <strong>Roll out gradually</strong> &mdash;
+                        See the <a href="{helpUrl}">Help tab</a> for a full rollout guide
+                        covering all enforcement levels and recovery procedures.
+                    </li>
+                </ol>
+            </f:be.infobox>
+        </f:if>
+
+        <!-- Configuration hints -->
+        <f:if condition="{configRpIdIsAutoDetected} && {configOriginIsAutoDetected} && {isNewInstallation} == 0">
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Resources/Private/Templates/AdminModule/Dashboard.html]: Fixed in 0a416bb. Changed to `!{isNewInstallation}`.

File: Resources/Private/Templates/AdminModule/Dashboard.html [hunk 5]
```diff
@@ -34,12 +79,28 @@ <h1><f:translate key="dashboard.title" extensionName="NrPasskeysBe" /></h1>
                         <span class="card-title" aria-label="{f:format.number(decimals: 1)}{adoptionPercentage}{/f:format.number}% {f:translate(key: 'dashboard.adoptionRate', extensionName: 'NrPasskeysBe')}">
                             <f:format.number decimals="1">{adoptionPercentage}</f:format.number>%
                         </span>
-                        <span class="card-subtitle"><f:translate key="dashboard.adoptionRate" extensionName="NrPasskeysBe" /></span>
+                        <span class="card-subtitle">
+                            <f:translate key="dashboard.adoptionRate" extensionName="NrPasskeysBe" />
+                            <span class="badge {adoptionBadge.class} ms-1">
+                                <core:icon identifier="{adoptionBadge.icon}" size="small" />
+                                {adoptionBadge.label}
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Resources/Private/Templates/AdminModule/Dashboard.html:86]: Fixed in 0a416bb. Badge labels now use `$this->translate()` with fallback English strings.

File: Resources/Private/Templates/AdminModule/Dashboard.html [hunk 6]
```diff
@@ -10,6 +10,51 @@ <h1><f:translate key="dashboard.title" extensionName="NrPasskeysBe" /></h1>
...
+                        covering all enforcement levels and recovery procedures.
+                    </li>
+                </ol>
+            </f:be.infobox>
+        </f:if>
+
+        <!-- Configuration hints -->
+        <f:if condition="{configRpIdIsAutoDetected} && {configOriginIsAutoDetected} && !{isNewInstallation}">
+            <f:be.infobox title="Configuration note" state="{f:constant(name: 'TYPO3\CMS\Fluid\ViewHelpers\Be\InfoboxViewHelper::STATE_WARNING')}">
+                Both Relying Party ID and Origin are auto-detected (currently <code>{configRpId}</code>).
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Resources/Private/Templates/AdminModule/Dashboard.html:52]: Fixed in 0a416bb. Changed to `!{isNewInstallation}`.

File: Resources/Private/Templates/AdminModule/Dashboard.html [hunk 7]
```diff
@@ -34,12 +79,28 @@ <h1><f:translate key="dashboard.title" extensionName="NrPasskeysBe" /></h1>
...
             </div>
         </div>
 
+        <!-- MFA hint: suggest dropping MFA when passkeys are adopted -->
+        <f:if condition="{usersWithPasskeys} > 0">
+            <f:be.infobox title="Passkeys &amp; MFA" state="{f:constant(name: 'TYPO3\CMS\Fluid\ViewHelpers\Be\InfoboxViewHelper::STATE_NOTICE')}">
+                Passkeys are inherently multi-factor (device + biometric/PIN) and phishing-resistant.
+                For most environments, passkeys alone are more secure than password + TOTP.
+                Consider disabling the MFA requirement for passkey-authenticated users.
+                <a href="{helpUrl}">Learn more in the Help tab.</a>
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Resources/Private/Templates/AdminModule/Dashboard.html:100]: The Getting Started and MFA hint texts are admin guidance that is unlikely to need localization in the near term. These are informational paragraphs, not UI labels. Will track as future i18n work if needed.

File: Classes/Controller/AdminModuleController.php [hunk 8]
```diff
@@ -164,4 +181,43 @@ private function getEnforcementLevelOptions(): array
...
+            return ['label' => $this->translate('dashboard.badge.noUsers', 'No users'), 'class' => 'badge-secondary', 'icon' => 'actions-minus'];
+        }
+
+        return match (true) {
+            $percentage >= 100.0 => ['label' => $this->translate('dashboard.badge.platinum', 'Platinum'), 'class' => 'badge-success', 'icon' => 'actions-bolt'],
+            $percentage >= 75.0 => ['label' => $this->translate('dashboard.badge.gold', 'Gold'), 'class' => 'badge-info', 'icon' => 'actions-star'],
+            $percentage >= 50.0 => ['label' => $this->translate('dashboard.badge.silver', 'Silver'), 'class' => 'badge-secondary', 'icon' => 'actions-check'],
+            $percentage >= 25.0 => ['label' => $this->translate('dashboard.badge.bronze', 'Bronze'), 'class' => 'badge-warning', 'icon' => 'actions-arrow-up'],
+            default => ['label' => $this->translate('dashboard.badge.gettingStarted', 'Getting started'), 'class' => 'badge-danger', 'icon' => 'actions-rocket'],
+        };
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Classes/Controller/AdminModuleController.php:221]: Fixed in 0a416bb. Labels use `$this->translate()` with XLIFF keys (dashboard.badge.platinum, etc.).

File: Classes/Controller/AdminModuleController.php [hunk 9]
```diff
@@ -77,13 +84,22 @@ public function dashboardAction(ServerRequestInterface $request): ResponseInterf
...
+        $config = $this->configService->getConfiguration();
+        $adoptionPercentage = $stats->adoptionPercentage();
+
         $moduleTemplate->assignMultiple([
             'totalUsers' => $stats->totalUsers,
             'usersWithPasskeys' => $stats->usersWithPasskeys,
-            'adoptionPercentage' => $stats->adoptionPercentage(),
+            'adoptionPercentage' => $adoptionPercentage,
+            'adoptionBadge' => $this->adoptionBadge($adoptionPercentage, $stats->totalUsers),
             'groups' => $groupData,
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Classes/Controller/AdminModuleController.php:95]: Added in 3a21098. 10 test cases via data provider covering all badge thresholds with boundary values (556 tests total now).

File: Classes/Controller/AdminModuleController.php [hunk 10]
```diff
@@ -77,13 +84,22 @@ public function dashboardAction(ServerRequestInterface $request): ResponseInterf
...
             'usersWithoutPasskeys' => $userData,
             'enforcementLevels' => $this->getEnforcementLevelOptions(),
+            'helpUrl' => (string) $this->uriBuilder->buildUriFromRoute('admin_passkeys.help'),
+            'configRpId' => $this->configService->getEffectiveRpId(),
+            'configRpIdIsAutoDetected' => $config->getRpId() === '',
+            'configOriginIsAutoDetected' => $config->getOrigin() === '',
+            'isNewInstallation' => $stats->usersWithPasskeys === 0,
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Classes/Controller/AdminModuleController.php:102]: Acknowledged. The name communicates the intent ("show Getting Started guide") rather than the implementation detail ("no passkeys yet"). Renaming to `hasNoPasskeys` would be more precise but less meaningful for the template condition. Keeping as-is for readability.

File: Resources/Private/Templates/AdminModule/Dashboard.html [hunk 11]
```diff
@@ -10,6 +10,51 @@ <h1><f:translate key="dashboard.title" extensionName="NrPasskeysBe" /></h1>
...
+                    <li class="mb-2">
+                        <strong>Set a pilot group to &quot;Encourage&quot;</strong> &mdash;
+                        Use the group table below to set a small group (e.g., your admin team)
+                        to <em>Encourage</em>. Users will see a dismissible banner prompting them
+                        to register a passkey.
+                    </li>
+                    <li>
+                        <strong>Roll out gradually</strong> &mdash;
+                        See the <a href="{helpUrl}">Help tab</a> for a full rollout guide
+                        covering all enforcement levels and recovery procedures.
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Resources/Private/Templates/AdminModule/Dashboard.html:43]: The Getting Started steps and MFA hint are admin guidance paragraphs, not reusable UI labels. Localizing these long-form texts would add significant XLIFF overhead for minimal benefit. Will track as future work if multi-language admin UIs are needed.
